tests/Unit/Value/CalcFunctionTest.php

```diff
@@ -0,0 +1,206 @@
+    public function provideValidOperatorSyntax(): array
+    {
+        return [
+            '+ op' => ['calc(100% + 20px)', 'calc(100% + 20px)'],
+            '- op' => ['calc(100% - 20px)', 'calc(100% - 20px)'],
+            '* op' => ['calc(100% * 20)', 'calc(100% * 20)'],
+            '* op no space' => ['calc(100%*20)', 'calc(100% * 20)'],
+            '/ op' => ['calc(100% / 20)', 'calc(100% / 20)'],
+            '/ op no space' => ['calc(100%/20)', 'calc(100% / 20)'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideValidOperatorSyntax
+     */
+    public function parseValidOperators(string $css, string $rendered): void
+    {
+        $calcFunction = $this->parse($css);
+        $output = $calcFunction->render(OutputFormat::create());
+        self::assertSame($rendered, $output);
+    }
+
+    /**
+     * @return array<string, array{0: string, 1: string}>
+     */
+    public function provideMultiline(): array
+    {
+        return [
+            'right newline' => ["calc(100% +\n20px)", 'calc(100% + 20px)'],
+            'right and outer newline' => ["calc(\n100% +\n20px\n)", 'calc(100% + 20px)'],
+            'left newline' => ["calc(100%\n+ 20px)", 'calc(100% + 20px)'],
+            'both newline' => ["calc(100%\n+\n20px)", 'calc(100% + 20px)'],
+            'tab whitespace' => ["calc(100%\t+\t20px)", 'calc(100% + 20px)'],
+            '- op' => ["calc(100%\n-\n20px)", 'calc(100% - 20px)'],
+            '/ op' => ["calc(100% /\n20)", 'calc(100% / 20)'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideMultiline
+     */
+    public function parseMultiline(string $css, string $rendered): void
+    {
+        $calcFunction = $this->parse($css);
+        $output = $calcFunction->render(OutputFormat::create());
+        self::assertSame($rendered, $output);
+    }
+
+    /**
+     * @return array<string, array{0: string}>
+     */
+    public function provideInvalidSyntax(): array
+    {
+        return [
+            'missing space around -' => ['calc(100%-20px)'],
+            'missing space around +' => ['calc(100%+20px)'],
+            'invalid operator' => ['calc(100% ^ 20px)'],
+        ];
+    }
+
+    /**
+     * @test
+     *
+     * @dataProvider provideInvalidSyntax
+     */
+    public function parseThrowsExceptionForInvalidSyntax(string $css): void
+    {
+        $this->expectException(UnexpectedTokenException::class);
+        $this->parse($css);
+    }
+
+    /**
+     * @test
+     */
+    public function parseThrowsExceptionIfCalledWithWrongFunctionName(): void
+    {
+        $css = 'wrong(100% - 20px)';
+        $parserState = new ParserState($css, Settings::create());
+
+        $this->expectException(UnexpectedTokenException::class);
+        $this->expectExceptionMessage('calc');
+        CalcFunction::parse($parserState);
+    }
+
+    /**
+     * Parse provided CSS as a CalcFunction
+     *
+     * @param string $css
+     * @return CalcFunction
```

Comment:
These PHPDoc annotations are actually redundant, since the types are in the method signature.